### PR TITLE
Add the dataset module from sticker

### PR DESCRIPTION
**Pull request by danieldk**
This could be imported without any changes, except for dropping some
mutability requirements.

Since many parts of this module were implemented by Tobias, this is
a good point to include the CONTRIBUTORS file.

### CODE REVIEW

File: CONTRIBUTORS [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Copyright 2018-2019 The sticker contributors
```
Comment (twuebi) [inline CONTRIBUTORS]: 2020?

File: sticker/src/dataset.rs [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        ConllxDataSet(read)
+    }
+
+    /// Returns an `Iterator` over `Result<Sentence, Error>`.
+    ///
+    /// Depending on the parameters the returned iterator filters
+    /// sentences by their lengths or returns the sentences in
+    /// sequence without filtering them.
+    ///
+    /// If `max_len` == `usize::MAX`, no filtering is performed.
```
Comment (twuebi) [inline sticker/src/dataset.rs]: If `max_len` == None, right?

Comment (twuebi) [reply sticker/src/dataset.rs]: Well, actually usize::MAX should have the same effect..

Comment (danieldk) [reply sticker/src/dataset.rs]: True :). But updated.

File: CONTRIBUTORS (hunk 1 above)
Comment (danieldk) [reply CONTRIBUTORS]: Fixed.

File: sticker/src/dataset.rs [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+            }
+            return Some(sent);
+        }
+        None
+    }
+}
+
+/// An Iterator adapter performing local shuffling.
+///
+/// Fills a buffer with size `buffer_size` on the first call. Subsequent
```
Comment (twuebi) [inline sticker/src/dataset.rs]: Subsequent calls add the next incoming item to the buffer and pick a random element from the buffer.

Comment (danieldk) [reply sticker/src/dataset.rs]: Updated.

**Review by twuebi — APPROVED** [review]

**Pull request merged**